i6_models/parts/conformer/mhsa_rel_pos.py

```diff
@@ -0,0 +1,248 @@
+    Conformer multi-headed self-attention module supporting
+        - relative positional encoding proposed by Shaw et al. (cf. https://arxiv.org/abs/1803.02155) by setting `learnable_pos_emb` to True and `with_pos_bias` to False
+        - and Transformer-XL style relative PE by Dai et al. (cf. https://arxiv.org/abs/1901.02860) by setting `learnable_pos_emb` to False and `with_pos_bias` to True
+
+    """
+
+    def __init__(self, cfg: ConformerMHSARelPosV1Config):
+
+        super().__init__()
+
+        self.layernorm = nn.LayerNorm(cfg.input_dim)
+
+        self.embed_dim = cfg.input_dim
+        self.num_heads = cfg.num_att_heads
+        self.embed_dim_per_head = self.embed_dim // self.num_heads
+
+        self.learnable_pos_emb = cfg.learnable_pos_emb
+        self.rel_pos_clip = cfg.rel_pos_clip
+        self.with_pos_bias = cfg.with_pos_bias
+        self.pos_emb_dropout = nn.Dropout(cfg.pos_emb_dropout)
+
+        assert not self.learnable_pos_emb or self.rel_pos_clip
+
+        self.att_weights_dropout = nn.Dropout(cfg.att_weights_dropout)
+
+        assert self.embed_dim % self.num_heads == 0, "embed_dim must be divisible by num_heads"
+
+        # projection matrices
+        self.q_proj_weight = nn.parameter.Parameter(torch.empty((self.embed_dim, self.embed_dim)))
+        self.k_proj_weight = nn.parameter.Parameter(torch.empty((self.embed_dim, self.embed_dim)))
+        self.v_proj_weight = nn.parameter.Parameter(torch.empty((self.embed_dim, self.embed_dim)))
+
+        self.in_proj_bias = nn.parameter.Parameter(torch.empty(3 * self.embed_dim))
+
+        self.out_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=True)
+
+        self.register_parameter("rel_pos_embeddings", None)
+        self.register_parameter("pos_bias_u", None)
+        self.register_parameter("pos_bias_v", None)
+
+        if self.learnable_pos_emb:
+            self.rel_pos_embeddings = nn.parameter.Parameter(
+                torch.empty(self.rel_pos_clip * 2 + 1, self.embed_dim // self.num_heads)
+            )
+        if self.with_pos_bias:
+            self.pos_bias_u = nn.parameter.Parameter(torch.empty(self.num_heads, self.embed_dim_per_head))
+            self.pos_bias_v = nn.parameter.Parameter(torch.empty(self.num_heads, self.embed_dim_per_head))
+
+        self.dropout = nn.Dropout1d(cfg.dropout) if cfg.dropout_broadcast_axes else nn.Dropout(cfg.dropout)
+        self.dropout_broadcast_axes = cfg.dropout_broadcast_axes
+
+        self._reset_parameters()  # initialize parameters
+
+    def _reset_parameters(self):
+        nn.init.xavier_uniform_(self.q_proj_weight)
+        nn.init.xavier_uniform_(self.k_proj_weight)
+        nn.init.xavier_uniform_(self.v_proj_weight)
+
+        # TODO: choose kind of initialization
+        if self.learnable_pos_emb:
+            nn.init.normal_(self.rel_pos_embeddings)
+        if self.with_pos_bias:
+            nn.init.constant_(self.pos_bias_u, 0.0)
+            nn.init.constant_(self.pos_bias_v, 0.0)
```

Comment:
I just checked, why do you do this different to ESPnet? In ESPnet, they do:
```python
        torch.nn.init.xavier_uniform_(self.pos_bias_u)
        torch.nn.init.xavier_uniform_(self.pos_bias_v)
```
So do the same?
```suggestion
            nn.init.xavier_uniform_(self.pos_bias_u)
            nn.init.xavier_uniform_(self.pos_bias_v)
```